### ISSUE TITLE
Databricks: `WITH SCHEMA EVOLUTION` syntax is not supported

### ISSUE THREAD

**Issue by cf-fshramko** [sqlfluff/sqlfluff#7042]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running SQLFluff against file with `WITH SCHEMA EVOLUTION` fails with parse error. 
For example:
```sql
CREATE VIEW IF NOT EXISTS
schema.table_name WITH SCHEMA EVOLUTION AS
select s.* from source_schema.source_table_name s;
```

### Expected Behaviour

Should correctly parse `WITH SCHEMA EVOLUTION` statement.

### Observed Behaviour

When I run `sqlfluff lint` the error is printed:
```
L:   5 | P:  17 |  PRS | Line 5, Position 16: Found unparsable section: 'WITH
                       | SCHEMA EVOLUTION AS\nwith r as (sele...'
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
```

### How to reproduce

Use the following sql:
```sql
CREATE VIEW IF NOT EXISTS
schema.table_name WITH SCHEMA EVOLUTION AS
select s.* from source_schema.source_table_name s;
```

and run `sqlfluff lint`

### Dialect

databricks

### Version

sqlfluff, version 3.4.2

### Configuration

`pyproject.toml` config:
```
[tool.sqlfluff.core]
templater = "placeholder"
dialect = "databricks"
max_line_length = 150
large_file_skip_byte_limit = 0

[tool.sqlfluff.layout.type.comma]
line_position = "leading"

[tool.sqlfluff.rules.references.special_chars]
quoted_identifiers_policy = "none"

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"

[tool.sqlfluff.rules.capitalisation.identifiers]
extended_capitalisation_policy = "snake"

[tool.sqlfluff.rules.ambiguous.column_references]
group_by_and_order_by_style = "explicit"

[tool.sqlfluff.templater.placeholder]
param_style = "dollar"
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
